### PR TITLE
[Diffusion][CPU] Adding AMX optimizations for CPU platform

### PR DESCRIPTION
**Pull request by jianan-gu**
This pr takes parts of follow-ups (mentioned in https://github.com/sgl-project/sglang/pull/20816) to bring key AMX based optimizations (scoping from LLM models) for CPU platforms, including:

- Adding CPU AMX backend, replace SDPA to bring best attention perf.
- Similar like for LLM models, adding replacement for amx weight packed linear. 
- Apply channel last 3d for VAE model

With above take Wan-AI/Wan2.2-TI2V-5B-Diffusers model as an example, the main stages show:

- TextEncodingStage speedup by 10x (bf16 encoding, if fp32 stays no change)
- DenoisingStage speedup by  2x
- DecodingStage speedup by 1.57x (bf16, enable channel last)

Besides, we still find more chance to bring more speedup (but with small ratio), by adding fusions (compile or kernel based) like rmsnorm/layernorm and newGELUs, will make this in followup PRs.

---
### CI States

Latest PR Test (Base): :x: [Run #28929873450](https://github.com/sgl-project/sglang/actions/runs/28929873450)
Latest PR Test (Extra): :x: [Run #28929873154](https://github.com/sgl-project/sglang/actions/runs/28929873154)

### CODE REVIEW

**Review by mingfeima — CHANGES_REQUESTED** [review]
some minor changes are needed.

File: test/registered/cpu/test_flash_attn.py [hunk 1]
```diff
@@ -138,6 +138,7 @@ def test_flash_attn_varlen(
             seqlens_q.max().item(),
             seqlens_k.max().item(),
             is_causal,
+            None,
```
Comment (mingfeima) [inline test/registered/cpu/test_flash_attn.py]: add test case when sm_scale is None and a given value.
some cases are not guarded in the test cases.

**Review by mingfeima — CHANGES_REQUESTED** [review]

File: test/registered/cpu/test_flash_attn.py (hunk 1 above)
Comment (jianan-gu) [reply test/registered/cpu/test_flash_attn.py]: sure, have added.

**Review by mingfeima — APPROVED** [review]

**Comment by mingfeima** [comment]
@jianan-gu rebase to fix conflict.

File: python/sglang/multimodal_gen/runtime/loader/fsdp_load.py [hunk 2]
```diff
@@ -336,6 +338,17 @@ def maybe_load_fsdp_model(
         # Avoid unintended computation graph accumulation during inference
         if isinstance(p, torch.nn.Parameter):
             p.requires_grad = False
+    local_torch_device = get_local_torch_device()
```
Comment (mickqian) [inline python/sglang/multimodal_gen/runtime/loader/fsdp_load.py:341]: we have an existing `quant_method.process_weights_after_loading` call in L321-L330, please avoid duplicating it

**Comment by mickqian** [comment]
this breaks CI, reverting in #30716

File: python/sglang/multimodal_gen/runtime/loader/fsdp_load.py (hunk 2 above)
Comment (jianan-gu) [reply python/sglang/multimodal_gen/runtime/loader/fsdp_load.py:341]: Thanks for pointing that, have submitted changes to  avoid such duplicating https://github.com/sgl-project/sglang/pull/30717
